core/services/gateway/api/jsonrpc/services/vault/service.go

```diff
@@ -0,0 +1,259 @@
+package vault
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api/jsonrpc"
+	gw_jsonrpc "github.com/smartcontractkit/chainlink/v2/core/services/gateway/api/jsonrpc"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/config"
+	gw_handlers "github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers"
+	hc "github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/common"
+)
+
+var (
+	ErrNotAllowlisted    = errors.New("sender not allowlisted")
+	ErrRateLimited       = errors.New("rate-limited")
+	ErrUnsupportedMethod = errors.New("unsupported method")
+
+	promHandlerError = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "gateway_vault_handler_error",
+		Help: "Metric to track vault handler errors",
+	}, []string{"don_id", "error"})
+
+	promSecretsCreateSuccess = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "gateway_vault_secrets_create_success",
+		Help: "Metric to track successful vault secrets_create calls",
+	}, []string{"don_id"})
+
+	promSecretsCreateFailure = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "gateway_vault_secrets_create_failure",
+		Help: "Metric to track failed vault secrets_create calls",
+	}, []string{"don_id"})
+)
+
+type service struct {
+	services.StateMachine
+
+	methodConfig VaultConfig
+	donConfig    *config.DONConfig
+	don          gw_handlers.DON
+	lggr         logger.Logger
+
+	mu                sync.RWMutex
+	userRateLimiter   *hc.RateLimiter
+	nodeRateLimiter   *hc.RateLimiter
+	requestTimeoutSec int
+
+	// In-memory secret storage for demo purposes
+	// In production, this would be a proper storage backend
+	secretsStore map[string]map[string]SecretEntry
+	storeMu      sync.RWMutex
+}
+
+type SecretEntry struct {
```

Comment:
As per the design doc, it also needs a namespace?
Also, I think another optional field LastUpdatedAt is good to add. It shows when was this secret last updated.

Or perhaps to be more advanced, it should be a UpdatedAt array field to list all the times it was updated at.